data/time_series_test.go

```diff
@@ -813,3 +814,782 @@ func TestFloatAt(t *testing.T) {
 		t.Errorf("Result mismatch (-want +got):\n%s", diff)
 	}
 }
+
+func TestFillMissing(t *testing.T) {
+	tests := []struct {
+		name        string
+		fillMissing data.FillMissing
+		frame       *data.Frame
+		expected    *data.Frame
+		Err         require.ErrorAssertionFunc
+	}{
+		{
+			name: "Long frame; fill previous",
+			fillMissing: data.FillMissing{
+				Enabled:  true,
+				Mode:     data.FILL_MODE_PREVIOUS,
+				Interval: 5 * time.Second,
+			},
+			frame: data.NewFrame("long_test",
+				data.NewField("Time", nil, []time.Time{
+					time.Date(2020, 1, 2, 3, 4, 0, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 20, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 29, 0, time.UTC),
+				}),
+				data.NewField("Values Floats", nil, []float64{
+					1.0,
+					3.0,
+					2.0,
+				}),
+				data.NewField("Values Int64", nil, []int64{
+					1,
+					3,
+					4,
+				}),
+				data.NewField("Animal Factor", nil, []string{
+					"cat",
+					"sloth",
+					"cat",
+				})),
+			expected: data.NewFrame("long_test",
+				data.NewField("Time", nil, []time.Time{
+					time.Date(2020, 1, 2, 3, 4, 0, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 5, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 10, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 15, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 20, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 25, 0, time.UTC),
+					time.Date(2020, 1, 2, 3, 4, 29, 0, time.UTC),
+				}),
+				data.NewField("Values Floats", nil, []float64{
+					1.0,
+					1.0,
+					1.0,
+					1.0,
+					3.0,
+					3.0,
+					2.0,
+				}),
+				data.NewField("Values Int64", nil, []int64{
+					1,
+					1,
+					1,
+					1,
+					3,
+					3,
+					4,
+				}),
+				data.NewField("Animal Factor", nil, []string{
```

Comment:
Perhaps this is me not understanding the feature. But wouldn't there be equal cat and sloth entries, since the missing entries for sloth would be filled?

Reply:
As I see it, the newly introduced rows are 1,2,3 and 5 corresponding to:
time.Date(2020, 1, 2, 3, 4, 5, 0, time.UTC),
time.Date(2020, 1, 2, 3, 4, 10, 0, time.UTC),
time.Date(2020, 1, 2, 3, 4, 15, 0, time.UTC),
time.Date(2020, 1, 2, 3, 4, 25, 0, time.UTC)
respectively and in the case of a non numeric fields I put the previous value
but probably this is not the desirable behaviour.